terraform/modules/vpc/main.tf

```diff
@@ -21,7 +22,7 @@ resource "aws_subnet" "public-subnets" {
   vpc_id                  = aws_vpc.vpc.id
   cidr_block              = var.PUBLIC_SUBNET[count.index]
   map_public_ip_on_launch = "true"
-  availability_zone       = data.aws_availability_zones.available.names[count.index]
+  availability_zone       = data.aws_availability_zones.available.names[count.index > length(data.aws_availability_zones.available.names) ? data.aws_availability_zones.available.names[count.index] : 0]
```

Comment:
This you try `null` instead of `0`? If yes, what was the error you were having?

Reply:
Error: Invalid index

  on modules/vpc/main.tf line 25, in resource "aws_subnet" "public-subnets":
  25:   availability_zone       = data.aws_availability_zones.available.names[count.index > length(data.aws_availability_zones.available.names) ? d                                             ata.aws_availability_zones.available.names[count.index] : null]
    |----------------
    | count.index is 0
    | data.aws_availability_zones.available.names is list of string with 6 elements

Can't use a null value as an indexing key.


Error: Invalid index

Reply:
@Nouman72884 You are putting condition block inside of an array that is why you can put null. This is not a good approach. Place the condition on start of the statement and use null instead of 0.